libs/balances-and-allowances/src/hooks/usePersistBalancesFromBff.ts

```diff
@@ -59,8 +90,15 @@ export function usePersistBalancesFromBff(params: PersistBalancesFromBffParams):
   }, [setBalances, isBalancesLoading, targetChainId, targetAccount])
 
   useEffect(() => {
+    const hasUnsupportedChainError = error instanceof Error && 
+      isUnsupportedChainError(error.message)
+    
+    if (hasUnsupportedChainError) {
+      addUnsupportedChainId(targetChainId)
+    }
+    
     setIsBffFailed(!!error)
-  }, [error, setIsBffFailed])
+  }, [error, setIsBffFailed, addUnsupportedChainId, targetChainId])
```

Comment:
Since `addUnsupportedChainId` is a part of hook dependencies, the callback must be wrapped in `useCallback`